tests/tagService_test.go

```diff
@@ -0,0 +1,355 @@
+		data       string
+		statusCode int
+		want       interface{}
+	}{
+		"simple": {
+			input:      1,
+			data:       `{"id": 1,"label": "TEST","color": "#1c71d8"}`,
+			statusCode: 200,
+			want: &gointelowl.Tag{
+				ID:    1,
+				Label: "TEST",
+				Color: "#1c71d8",
+			},
+		},
+		"cantFind": {
+			input:      9000,
+			data:       `{"detail": "Not found."}`,
+			statusCode: 404,
+			want: &gointelowl.IntelOwlError{
+				StatusCode: 404,
+				Data:       []byte(`{"detail": "Not found."}`),
+			},
+		},
+	}
+	for name, testCase := range testCases {
+		//* Subtest
+		t.Run(name, func(t *testing.T) {
+			testServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(testCase.statusCode)
+				_, writeError := w.Write([]byte(testCase.data))
+				if writeError != nil {
+					fmt.Printf("Error: %s", writeError)
+				}
+			}))
+			defer testServer.Close()
+			client := gointelowl.MakeNewIntelOwlClient(
+				&gointelowl.IntelOwlClientOptions{
+					Url:         testServer.URL,
+					Token:       "test-token",
+					Certificate: "",
+				},
+				nil,
+			)
+			ctx := context.Background()
+			gottenTag, err := client.TagService.Get(ctx, testCase.input)
+			if testCase.statusCode < http.StatusOK || testCase.statusCode >= http.StatusBadRequest {
+				diff := cmp.Diff(testCase.want, err)
+				if diff != "" {
+					t.Fatalf(diff)
+				}
+			} else {
+				diff := cmp.Diff(testCase.want, gottenTag)
+				if diff != "" {
+					t.Fatalf(diff)
+				}
+			}
+		})
+	}
+}
+
+func TestTagServiceCreate(t *testing.T) {
+	// *table test case
+	testCases := map[string]struct {
+		input      *gointelowl.TagParams
+		data       string
+		statusCode int
+		want       interface{}
+	}{
+		"simple": {
+			input: &gointelowl.TagParams{
+				Label: "TEST TAG",
+				Color: "#fffff",
+			},
+			data:       `{"id": 1,"label": "TEST TAG","color": "#fffff"}`,
+			statusCode: 200,
+			want: &gointelowl.Tag{
+				ID:    1,
+				Label: "TEST TAG",
+				Color: "#fffff",
+			},
+		},
+		"duplicate": {
+			input: &gointelowl.TagParams{
+				Label: "TEST TAG",
+				Color: "#fffff",
+			},
+			data:       `{"label":["tag with this label already exists."]}`,
+			statusCode: 400,
+			want: &gointelowl.IntelOwlError{
+				StatusCode: 400,
+				Data:       []byte(`{"label":["tag with this label already exists."]}`),
+			},
+		},
+	}
+	for name, testCase := range testCases {
+		//* Subtest
+		t.Run(name, func(t *testing.T) {
+			testServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(testCase.statusCode)
+				_, err := w.Write([]byte(testCase.data))
+				if err != nil {
+					fmt.Printf("Error: %s", err)
+				}
+
+			}))
+			defer testServer.Close()
+			client := gointelowl.MakeNewIntelOwlClient(
+				&gointelowl.IntelOwlClientOptions{
+					Url:         testServer.URL,
+					Token:       "test-token",
+					Certificate: "",
+				},
+				nil,
+			)
+			ctx := context.Background()
```

Comment:
This is also duplicated across each test-case method. This should be extracted as a fixture or something on the global/file level and make it available to each test-case method.

Reply:
So basically make a function like `makeNewTestServer` that takes parameters of what will the test server will output and return  a new `httptest.NewServer` and make a function for the `IntelOwlClient`.